diff --git a/qtbase/src/corelib/io/qprocess_unix.cpp b/qtbase/src/corelib/io/qprocess_unix.cpp
index 0000000000..0000000000 100644
--- a/qtbase/src/corelib/io/qprocess_unix.cpp
+++ b/qtbase/src/corelib/io/qprocess_unix.cpp
@@ -342,7 +342,7 @@ private:
         Q_UNREACHABLE_RETURN(-1);
     }
 
-#if defined(PTHREAD_CANCEL_DISABLE)
+#if defined(PTHREAD_CANCEL_DISABLE) && !defined(OPENHARMONY) && !defined(__OHOS__)
     int oldstate = PTHREAD_CANCEL_DISABLE;
     void disableThreadCancellations() noexcept
     {
diff --git a/qtbase/src/corelib/thread/qthread_unix.cpp b/qtbase/src/corelib/thread/qthread_unix.cpp
index 0000000000..0000000000 100644
--- a/qtbase/src/corelib/thread/qthread_unix.cpp
+++ b/qtbase/src/corelib/thread/qthread_unix.cpp
@@ -326,7 +326,7 @@ void terminate_on_exception(T &&t)
 
 void *QThreadPrivate::start(void *arg)
 {
-#ifdef PTHREAD_CANCEL_DISABLE
+#if defined(PTHREAD_CANCEL_DISABLE) && !defined(OPENHARMONY) && !defined(__OHOS__)
     pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, nullptr);
 #endif
     QThread *thr = reinterpret_cast<QThread *>(arg);
@@ -368,7 +368,7 @@ void *QThreadPrivate::start(void *arg)
 #endif
 
         emit thr->started(QThread::QPrivateSignal());
-#ifdef PTHREAD_CANCEL_DISABLE
+#if defined(PTHREAD_CANCEL_DISABLE) && !defined(OPENHARMONY) && !defined(__OHOS__)
         pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, nullptr);
         pthread_testcancel();
 #endif
@@ -390,7 +390,7 @@ void QThreadPrivate::finish()
         // Disable cancellation; we're already in the finishing touches of this
         // thread, and we don't want cleanup to be disturbed by
         // abi::__forced_unwind being thrown from all kinds of functions.
-#ifdef PTHREAD_CANCEL_DISABLE
+#if defined(PTHREAD_CANCEL_DISABLE) && !defined(OPENHARMONY) && !defined(__OHOS__)
         pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, nullptr);
 #endif
 
@@ -417,7 +417,7 @@ void QThreadPrivate::cleanup()
 
         // Disable cancellation again: we did it above, but some user code
         // running between finish() and cleanup() may have turned them back on.
-#ifdef PTHREAD_CANCEL_DISABLE
+#if defined(PTHREAD_CANCEL_DISABLE) && !defined(OPENHARMONY) && !defined(__OHOS__)
         pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, nullptr);
 #endif
 
@@ -816,7 +816,7 @@ void QThread::start(Priority priority)
 
 void QThread::terminate()
 {
-#if !defined(Q_OS_ANDROID)
+#if !defined(Q_OS_ANDROID) && !defined(OPENHARMONY) && !defined(__OHOS__)
     Q_D(QThread);
     QMutexLocker locker(&d->mutex);
 
@@ -879,7 +879,7 @@ void QThread::setTerminationEnabled(bool enabled)
                "Current thread was not started with QThread.");
 
     Q_UNUSED(thr);
-#if defined(Q_OS_ANDROID)
+#if defined(Q_OS_ANDROID) || defined(OPENHARMONY) || defined(__OHOS__)
     Q_UNUSED(enabled);
 #else
     pthread_setcancelstate(enabled ? PTHREAD_CANCEL_ENABLE : PTHREAD_CANCEL_DISABLE, nullptr);
